File: ProjectEulerPython/problems/problem_020.py
from problems.problem import Problem

class Problem020(Problem):

  def calculate_answer(self) -> int:
    n = 100
    product = 1
    for i in range(2, n + 1):
      product *= i

    self.print_detail("Product:")
    self.print_detail(product)

    answer = 0
    product_str = str(product)
    for i in range(0, len(product_str)):
      answer += int(product_str[i])

    return answer

File: ProjectEuler/Problems/Problem020.cs
using Microsoft.Extensions.Logging;

namespace ProjectEuler.Problems
{
    public class Problem020 : Problem<int>
    {
        public Problem020(ILogger<Problem020> logger) : base(logger) { }

        protected override int CalculateAnswer()
        {
            var product = "1";
            for (var i = 2; i <= 100; ++i)
            {
                product = MathHelpers.MultiplyIntegerStrings(product, i.ToString());
            }

            Logger.LogDebug("{product}", product);

            var answer = 0;
            for (var i = 0; i < product.Length; ++i)
            {
                answer += product[i] - '0';
            }

            return answer;
        }
    }
}
